plugins/core/IO/qAdditionalIO/src/PVFilter.cpp

```diff
@@ -181,41 +182,40 @@ CC_FILE_ERROR PVFilter::loadFile(const QString& filename, ccHObject& container,
 
 	for (unsigned i = 0; i < numberOfPoints; i++)
 	{
-		//if we reach the max. cloud size limit, we cerate a new chunk
 		if (pointsRead == fileChunkPos+fileChunkSize)
 		{
+			//if we reach the max. cloud size limit, we create a new chunk
 			if (loadedCloud)
 			{
-				int sfIdx = loadedCloud->getCurrentInScalarFieldIndex();
-				if (sfIdx>=0)
+				if (sf)
 				{
-					CCCoreLib::ScalarField* sf = loadedCloud->getScalarField(sfIdx);
 					sf->computeMinAndMax();
-					loadedCloud->setCurrentDisplayedScalarField(sfIdx);
+					loadedCloud->setCurrentDisplayedScalarField(loadedCloud->getCurrentInScalarFieldIndex());
 					loadedCloud->showSF(true);
 				}
 				container.addChild(loadedCloud);
 			}
 			fileChunkPos = pointsRead;
 			fileChunkSize = std::min<unsigned>(numberOfPoints - pointsRead, CC_MAX_NUMBER_OF_POINTS_PER_CLOUD);
 			loadedCloud = new ccPointCloud(QString("unnamed - Cloud #%1").arg(++chunkIndex));
+			sf = nullptr;
 			if (!loadedCloud || !loadedCloud->reserveThePointsTable(fileChunkSize) || !loadedCloud->enableScalarField())
 			{
 				result = CC_FERR_NOT_ENOUGH_MEMORY;
 				delete loadedCloud;
 				loadedCloud = nullptr;
 				break;
 			}
+			sf = loadedCloud->getCurrentInScalarField();
+			assert(sf);
 		}
 
 		//we read the 3 coordinates of the point
```

Comment:
I got distracted the last month or so and forgot to send up a PR for the PVfilter changes, I did mostly the same fix, I thought I'd post here so you can decide which you prefer, I think it simplifies things reading 4 values in at once so you don't have to do a NAN fill

```C++
//we read the 3 coordinates of the point and the scalar
		float rBuff[4];
		if (in.read((char*)rBuff, 4 * sizeof(float)) >= 0)
		{
			//conversion to CCVector3
			CCVector3 P((PointCoordinateType)rBuff[0],
						(PointCoordinateType)rBuff[1],
						(PointCoordinateType)rBuff[2]);
			loadedCloud->addPoint(P);
			loadedSF->addElement((ScalarType)rBuff[3]);
		}
		else
		{
			result = CC_FERR_READING;
			break;
		}
```

Reply:
Yes, definitely simpler!